stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/LoadToTable.java

```diff
@@ -0,0 +1,164 @@
+import org.ballerinalang.model.values.BTable;
+import org.ballerinalang.model.values.BTypeValue;
+import org.ballerinalang.nativeimpl.io.channels.FileIOChannel;
+import org.ballerinalang.nativeimpl.io.channels.base.CharacterChannel;
+import org.ballerinalang.nativeimpl.io.channels.base.DelimitedRecordChannel;
+import org.ballerinalang.nativeimpl.io.utils.IOUtils;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Native function ballerina.io#loadToTable.
+ *
+ * @since 0.964.1
+ */
+@BallerinaFunction(
+        packageName = "ballerina.io",
+        functionName = "loadToTable",
+        args = {@Argument(name = "path", type = TypeKind.STRING),
+                @Argument(name = "recordSeparator", type = TypeKind.STRING),
+                @Argument(name = "fieldSeparator", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.TABLE),
+                      @ReturnType(type = TypeKind.STRUCT, structType = "IOError", structPackage = "ballerina.io")},
+        isPublic = true
+)
+public class LoadToTable extends BlockingNativeCallableUnit {
+
+    private static final Logger log = LoggerFactory.getLogger(LoadToTable.class);
+
+    @Override
+    public void execute(Context context) {
+        final String filePath = context.getStringArgument(0);
+        Path path;
+        try {
+            path = Paths.get(filePath);
+        } catch (InvalidPathException e) {
+            String msg = "Unable to resolve the file path: " + filePath;
+            log.error(msg, e);
+            context.setReturnValues(new BTable(), IOUtils.createError(context, msg));
+            return;
+        }
+        if (Files.notExists(path)) {
+            String msg = "Unable to find a file in given path: " + filePath;
+            context.setReturnValues(new BTable(), IOUtils.createError(context, msg));
+            return;
+        }
+        try {
+            BTable table = getbTable(context, path);
+            context.setReturnValues(table, null);
+        } catch (IOException e) {
+            String msg = "Failed to process the delimited file: " + e.getMessage();
+            log.error(msg, e);
+            context.setReturnValues(new BTable(), IOUtils.createError(context, msg));
+        }
+    }
+
+    private BTable getbTable(Context context, Path path) throws IOException, BallerinaIOException {
+        BTypeValue type = (BTypeValue) context.getRefArgument(0);
+        if (type == null) {
```

Comment:
`context.getRefArgument(0)` will throw a null reference error, if the value is null. Hence we don't need this null check.
If we need to throw a meaningful error, we can use `context.getNullableRefArgument(0)`, and do the  null check and throw a meaningful error.